content/docs/higher-order-components.md

```diff
@@ -1,56 +1,56 @@
 ---
 id: higher-order-components
-title: Higher-Order Components
+title: Komponenty wyższego rzędu
 permalink: docs/higher-order-components.html
 ---
 
-A higher-order component (HOC) is an advanced technique in React for reusing component logic. HOCs are not part of the React API, per se. They are a pattern that emerges from React's compositional nature.
+Komponent wyższego rzędu (ang. *Higher-Order Component*), w skrócie KWR (ang. *HOC*), to zaawansowana technika reactowa stosowana w celu wielokrotnego używania logiki komponentu. KWR-y nie są częścią API Reacta *per se*. Są wzorcem, który wyłonił się z kompozycyjnej natury Reacta.
 
-Concretely, **a higher-order component is a function that takes a component and returns a new component.**
+Konkretnie rzecz ujmując, **komponent wyższego rzędu jest funkcją, która przyjmuje jako argument inny komponent i zwraca nowy komponent.**
 
 ```js
 const EnhancedComponent = higherOrderComponent(WrappedComponent);
 ```
 
-Whereas a component transforms props into UI, a higher-order component transforms a component into another component.
+Tak jak zwykły komponent przekształca właściwości (ang. *props*) na fragment UI, tak komponent wyższego rzędu przekształca komponent w inny komponent.
 
-HOCs are common in third-party React libraries, such as Redux's [`connect`](https://github.com/reduxjs/react-redux/blob/master/docs/api/connect.md#connect) and Relay's [`createFragmentContainer`](http://facebook.github.io/relay/docs/en/fragment-container.html).
+KWR-y pojawiają się często w zewnętrznych bibliotekach reactowych, np. [`connect`](https://github.com/reduxjs/react-redux/blob/master/docs/api/connect.md#connect) w Reduksie czy [`createFragmentContainer`](http://facebook.github.io/relay/docs/en/fragment-container.html) w Relayu.
 
-In this document, we'll discuss why higher-order components are useful, and how to write your own.
+W tym artykule podejmiemy dyskusję o tym, dlaczego komponenty wyższego rzędu są użyteczne oraz jak napisać własny.
```

Comment:
"podejmiemy dyskusję" to chyba zbyt brutalna kalka językowa ;) może coś bardziej lajtowego jak "wyjaśnimy"?